### ISSUE TITLE
Support interaction with godep

### ISSUE THREAD

**Issue by eapache** [mailgun/godebug#15]
A lot of go programs use https://github.com/tools/godep for dependency management, which also provides a "wrapper" command (godep go build, godep go test, etc) for manipulating `GOPATH`. Dunno if this is possible, but it would be very handy to be able to godebug projects that use godep.

**Comment by jeremyschlatter** [comment]
Good point, thanks! godebug could have an option to use an arbitrary wrapper instead of the go command. The two tricky parts that come to mind now are: (1) The initial parsing and type-checking of the program, for which there is probably no godep wrapper command, and (2) Multiple tools manipulating `GOPATH` could easily run into problems [related](https://github.com/mailgun/godebug/commit/e5ba457312a994ebe0adbecfbc1d55769035fbf6) to [this](https://github.com/golang/go/blob/9d0239771a2ddd77be0ba64c2782e1328a378190/src/cmd/go/pkg.go#L805-L811).

I'll look into this more.

**Comment by shurcooL** [comment]
> easily run into problems [related](https://github.com/mailgun/godebug/commit/e5ba457312a994ebe0adbecfbc1d55769035fbf6) to [this](https://github.com/golang/go/blob/9d0239771a2ddd77be0ba64c2782e1328a378190/src/cmd/go/pkg.go#L805-L811).

The latter is being addressed for Go 1.5 in https://github.com/golang/go/issues/10509! \o/

**Comment by jeremyschlatter** [comment]
> The latter is being addressed for Go 1.5 in golang/go#10509! \o/

Nice! That's awesome.
